File: src/main/java/org/mitallast/queue/common/component/LifecycleComponent.java
package org.mitallast.queue.common.component;

public interface LifecycleComponent {

    Lifecycle lifecycle();

    void checkIsStarted() throws IllegalStateException;

    void checkIsStopped() throws IllegalStateException;

    void checkIsClosed() throws IllegalStateException;

    void start();

    void stop();

    void close();
}

File: src/main/java/org/mitallast/queue/common/component/LifecycleComponent.kt
package org.mitallast.queue.common.component

interface LifecycleComponent {

    fun lifecycle(): Lifecycle

    @Throws(IllegalStateException::class)
    fun checkIsStarted()

    @Throws(IllegalStateException::class)
    fun checkIsStopped()

    @Throws(IllegalStateException::class)
    fun checkIsClosed()

    fun start()

    fun stop()

    fun close()
}
